app.rb

```diff
@@ -0,0 +1,142 @@
+require 'date'
+require_relative 'library/classes/person'
+require_relative 'library/classes/teacher'
+require_relative 'library/classes/student'
+require_relative 'library/classes/book'
+require_relative 'library/classes/rental'
+
+# This class represents the main application for managing books, people, and rentals.
+
+# rubocop:disable Metrics
+class App
+  def display_menu
+    puts 'Please select an option by entering a number'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a Book'
+    puts '5. Create a Rental'
+    puts '6. List all rentals for a given person id'
+    puts '7. Exit'
+  end
+
+  def run
+    loop do
+      books_count = Book.books_count
+      Person.people_count
+      Rental.rentals_count
+      students_count = Student.students_count
+      teachers_count = Teacher.teachers_count
+
+      display_menu
+
+      # choice
+      choice = gets.chomp.to_i
+
+      # case choice
+      case choice
+      when 1
+        puts "We have #{books_count} book."
+        if books_count.positive?
+          puts 'Here is the list of all books'
+          Book.list_all_books
+        end
+      when 2
+        puts "[Students] (#{students_count})"
+        Student.list_all_students
+        puts "[Teachers] (#{teachers_count})"
+        Teacher.list_all_teachers
+      when 3
+        # Create a person
+        print 'Do you want to create a student(1) or a teacher(2)? [input the number]: '
+        input = gets.chomp.to_i
+        print 'Age: '
+        age = gets.chomp.to_i
+        print 'Name: '
+        name = gets.chomp
+        if input == 1
+          print 'Classroom: '
+          classroom = gets.chomp
+          Student.add_a_student(age, name, classroom)
+          puts "The student #{name} has been created and added to the classroom #{classroom}"
+        elsif input == 2
```

Comment:
- [ ] From the video guide, when you chose a student on option `1`, there's a parent permission option which yours is missing, instead a classroom option was asked. Kindly fix that 👍 

### Vidoe prompt:
![image](https://github.com/diarisdiakite/oop_school-library/assets/50990925/8740e2de-4cbd-47df-84f8-d46fab4cd3e9)

### Your prompt:
![image](https://github.com/diarisdiakite/oop_school-library/assets/50990925/8c086d59-6059-4794-b619-09befdd8f84e)

Reply:
Done @Shedrack-Sunday, Thanks.